packages/react-reconciler/src/ReactFiberLane.js

```diff
@@ -7,16 +7,677 @@
+        UpdateRangeEnd,
+        wipLanes,
+      );
+      if (lane === NoLane) {
+        lane = InputDiscreteHydrationLane;
+      }
+      return lane;
+    }
+    case InputContinuousLanePriority: {
+      let lane = findLane(
+        InputContinuousUpdateRangeStart,
+        UpdateRangeEnd,
+        wipLanes,
+      );
+      if (lane === NoLane) {
+        lane = InputContinuousHydrationLane;
+      }
+      return lane;
+    }
+    case DefaultLanePriority: {
+      let lane = findLane(DefaultUpdateRangeStart, UpdateRangeEnd, wipLanes);
+      if (lane === NoLane) {
+        lane = DefaultHydrationLane;
+      }
+      return lane;
+    }
+    case TransitionShortLanePriority:
+    case TransitionLongLanePriority:
+      // Should be handled by findTransitionLane instead
+      break;
+    case IdleLanePriority:
+      let lane = findLane(IdleUpdateRangeStart, IdleUpdateRangeEnd, IdleLanes);
+      if (lane === NoLane) {
+        lane = IdleHydrationLane;
+      }
+      return lane;
+    default:
+      // The remaining priorities are not valid for updates
+      break;
+  }
+  invariant(
+    false,
+    'Invalid update priority:  %s. This is a bug in React.',
+    lanePriority,
+  );
+}
+
+// To ensure consistency across multiple updates in the same event, this should
+// be pure function, so that it always returns the same lane for given inputs.
+export function findTransitionLane(
+  lanePriority: LanePriority,
+  wipLanes: Lanes,
+  pendingLanes: Lanes,
+): Lane {
+  if (lanePriority === TransitionShortLanePriority) {
+    let lane = findLane(
+      TransitionShortUpdateRangeStart,
+      TransitionShortUpdateRangeEnd,
+      wipLanes | pendingLanes,
+    );
+    if (lane === NoLane) {
+      lane = findLane(
+        TransitionShortUpdateRangeStart,
+        TransitionShortUpdateRangeEnd,
+        wipLanes,
+      );
+      if (lane === NoLane) {
+        lane = TransitionShortHydrationLane;
+      }
+    }
+    return lane;
+  }
+  if (lanePriority === TransitionLongLanePriority) {
+    let lane = findLane(
+      TransitionLongUpdateRangeStart,
+      TransitionLongUpdateRangeEnd,
+      wipLanes | pendingLanes,
+    );
+    if (lane === NoLane) {
+      lane = findLane(
+        TransitionLongUpdateRangeStart,
+        TransitionLongUpdateRangeEnd,
+        wipLanes,
+      );
+      if (lane === NoLane) {
+        lane = TransitionLongHydrationLane;
+      }
+    }
+    return lane;
+  }
+  invariant(
+    false,
+    'Invalid transition priority:  %s. This is a bug in React.',
+    lanePriority,
+  );
+}
+
+function findLane(start, end, skipLanes) {
+  let lane = 1 << start;
+  let i = start;
+  while (i < end) {
+    if ((skipLanes & lane) === NoLane) {
+      return lane;
+    }
+    i++;
+    lane <<= 1;
+  }
+  return NoLane;
+}
+
+function getLowestPriorityLane(lanes: Lanes): Lane {
+  // This finds the most significant non-zero bit.
+  // TODO: Consider alternate implementations. Could use a map. Is Math.log2
```

Comment:
I think `Math.clz32` could work!

Reply:
Oh I didn’t realize so many browsers supported this. We can use a fallback when it’s not there. Good idea, thanks!